src/bitcoin-cli.cpp

```diff
@@ -867,6 +875,99 @@ static void GetWalletBalances(UniValue& result)
     result.pushKV("balances", balances);
 }
 
+/**
+ * ParseGetInfoResult takes in -getinfo result in UniValue object and parses it
+ * into a user friendly UniValue string to be printed on the console.
+ * @param[out] result  Reference to UniValue result containing the -getinfo output.
+ */
+static void ParseGetInfoResult(UniValue& result)
+{
+    if (!find_value(result, "error").isNull()) return;
+
+    std::string RESET, GREEN, BLUE, YELLOW, MAGENTA, CYAN;
+    bool should_colorize = false;
+
+#ifndef WIN32
+    if (isatty(fileno(stdout))) {
+        // By default, only print colored text if OS is not WIN32 and stdout is connected to a terminal.
+        should_colorize = true;
+    }
+#endif
+
+    if (gArgs.IsArgSet("-color")) {
+        // Override default should_colorize value if -color is set.
+        std::string color = gArgs.GetArg("-color", "auto");
```

Comment:
The idea is that when somebody passes `-color=nonsense`, then you consider the value to be `auto` (no warning or so), right?

Reply:
Yup! If they don't enter a valid "when" value, it would be treated as the default colouring.